src/exception.jl

```diff
@@ -132,7 +132,7 @@ be called via Python C-API, it tries to not throw at all cost.
 function showerror_string(e::T) where {T}
     try
         io = IOBuffer()
-        showerror(io, e)
+        showerror(io, e, catch_backtrace())
```

Comment:
I'm worried that this isn't valid  We should really only call `catch_backtrace()` in a `catch` block, otherwise its value may be unpredictable.

Can the caller to `showerror_string` pass in the backtrace if it is available?